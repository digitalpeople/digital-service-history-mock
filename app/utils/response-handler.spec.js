import responseHandler from './response-handler';

describe('responseHandler', () => {
  it('should handle if no error happens', () => {
    expect.assertions(1);

    const error = undefined;
    const data = {
      foo: 'bar',
    };
    const response = {
      send: jest.fn(),
      sendStatus: jest.fn(),
    };

    responseHandler(response)(error, data);

    expect(response.send.mock.calls[0][0]).toMatchSnapshot();
  });

  it('should handle if any error happens', () => {
    expect.assertions(1);

    const error = {
      message: 'error-message',
    };
    const data = {
      foo: 'bar',
    };
    const response = {
      send: jest.fn(),
      sendStatus: jest.fn(),
    };

    responseHandler(response)(error, data);

    expect(response.sendStatus.mock.calls[0][0]).toMatchSnapshot();
  });
});
