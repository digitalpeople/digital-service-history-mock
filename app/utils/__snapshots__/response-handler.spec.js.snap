// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responseHandler should handle if any error happens 1`] = `404`;

exports[`responseHandler should handle if no error happens 1`] = `
Object {
  "foo": "bar",
}
`;
